### PR TITLE
test: add more tests for pkg/app/middlewares/server/basic_auth

### PR DESCRIPTION
**Pull request by mamil**
#### What type of PR is this?
test: Adding missing tests or correcting existing tests

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
为pkg/app/middlewares/server/basic_auth添加更多单元测试

#### (Optional) More detail description for this PR(en: English/zh: Chinese).
en: Add BasicAuth related tests in pkg/app/middlewares/server/basic_auth, basic_auth.go reaches 100% coverage
zh(optional): 添加pkg/app/middlewares/server/basic_auth中BasicAuth相关的测试，basic_auth.go达到100%覆盖

#### Which issue(s) this PR fixes:
#257

### CODE REVIEW

**Comment by welkeyever** [comment]
Please resign the CLA🙏

**Comment by mamil** [comment]
@welkeyever CLA signed

File: pkg/app/middlewares/server/basic_auth/basic_auth_test.go [hunk 1]
```diff
@@ -63,3 +67,32 @@ func TestPairs(t *testing.T) {
...
+
+	c2 := app.RequestContext{}
+	encodeStr = "Basic " + base64.StdEncoding.EncodeToString(bytesconv.S2b(userName2+":"+password2))
+	c2.Request.Header.Add("Authorization", encodeStr)
+
+	handler(context.TODO(), &c2)
+
+	user, ok = c2.Get("user")
+	assert.DeepEqual(t, nil, user)
+	assert.DeepEqual(t, false, ok)
```
Comment (welkeyever) [inline pkg/app/middlewares/server/basic_auth/basic_auth_test.go:97]: Better to use `assert.False` and `assert.True`, keep the style consistent with others.

Comment (mamil) [reply pkg/app/middlewares/server/basic_auth/basic_auth_test.go]: done

**Review by welkeyever — APPROVED** [review]
LGTM, thx~

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**